src/app/Http/Controllers/Api/MovieController.php

```diff
@@ -13,7 +13,10 @@ class MovieController extends Controller
 
     public function index(Request $request)
     {
-        return new MovieCollection(Movie::paginate(8));
+        $movies = Movie::paginate(8);
+        if($request['genre'] >= 0) $movies = $movies->where('genre', $request['genre']);
```

Comment:
Prvo filtriras pa onda radis paginaciju

$query = Movie::all();
if($request->filled('genre'){
    $query->where('genre', $request->genre);
}
return MovieResource::collection($query->paginate(8));
Ovako nesto

Reply:
Pa onda unutar MovieResource uradis nesto ovako:

return [
    "id" => $this->id,
     ..... ,
    ....... ,
   "genre" => new GenreResource($this->whenLoaded('genre')),
];

Ovo ce ocitati genre samo ako je eager loadovan inace nece poslati nazad ka FE